From: Hannes Reinecke <hare@suse.de>
Date: Fri, 28 Feb 2014 12:41:10 +0100
Subject: scsi_dh_alua: fixup RTPG retry delay miscalculation
References: bnc#854025
Patch-Mainline: Not yet

The calculation for the workqueue delay was wrong, resulting
in delayed initialisation.

Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/device_handler/scsi_dh_alua.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/scsi/device_handler/scsi_dh_alua.c b/drivers/scsi/device_handler/scsi_dh_alua.c
index 87f1986..54e60e7 100644
--- a/drivers/scsi/device_handler/scsi_dh_alua.c
+++ b/drivers/scsi/device_handler/scsi_dh_alua.c
@@ -966,10 +966,10 @@ static void alua_rtpg_work(struct work_struct *work)
 		pg->flags &= ~ALUA_PG_RUN_STPG;
 		if (err == SCSI_DH_RETRY) {
 			pg->flags |= ALUA_PG_RUN_RTPG;
-			pg->interval = ALUA_RTPG_DELAY_MSECS * 1000;
+			pg->interval = 0;
 			spin_unlock_irqrestore(&pg->rtpg_lock, flags);
 			queue_delayed_work(kmpath_aluad, &pg->rtpg_work,
-					   pg->interval * HZ);
+				msecs_to_jiffies(ALUA_RTPG_DELAY_MSECS));
 			return;
 		}
 	}
-- 
1.7.12.4

